src/ui/profile/ProfileSelection.js

```diff
@@ -0,0 +1,170 @@
+// eslint-disable-next-line no-redeclare
+/* global chrome */
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import styles from '../styles/profile.module.css';
+import AddProfile from './AddProfile';
+import BeyondLoader from './BeyondLoader';
+import Profiles from './Profile';
+
+const LoadedProfile = (profileModel) => ({
+  status: 'loaded',
+  profile: profileModel,
+});
+
+const ErrorProfile = (error, profile) => ({
+  status: 'error',
+  error,
+  profile,
+});
+
+const LoadingProfile = (profile) => ({
+  status: 'loading',
+  profile,
+});
+
+function ProfileSelection(props) {
+  const { onCharacterReady } = props;
+  const [profiles, setProfiles] = useState(new Map());
+  const [currentProfile, setCurrentProfile] = useState(null);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  useEffect(() => {
+    chrome.storage.local.get('savedProfiles', (result) => {
```

Comment:
Just a thought exercise, but this looks like the perfect place for a custom hook. You could call it `useChromeLocalStorage()`:

```js
function useChromeLocalStorage(storageKey, defaultValue) {
    const [state, setState] = useState(defaultValue);

    useEffect(() => {
        chrome.storage.local.get(storageKey, result => {
            if (!result[storageKey]) {
                chrome.storage.local.set({ [storageKey]: defaultValue });
            } else {
                setState(result[storageKey]);
            }
        };
    }, []);

    const setStorageValue = value => {
        setState(value);
        chrome.storage.local.set({ [storageKey]: value });
    };

    return [state, setStorageValue];
}
```

It acts similarly to `useState()`, but it also keeps the state automatically in sync with the chrome storage.

This is one of the beautiful things about hooks: composability.

Reply:
If you were to use it here, it would look like this:

```js
const [profiles, setProfiles] = useChromeLocalStorage('savedProfiles', {});
```

And you wouldn't need this `useEffect` here.

Reply:
If you decide not to use it (which is of course fine), this `useEffect()` would still need to be changed because you're using an object now:

```js
useEffect(() => {
    chrome.storage.local.get('savedProfiles', result => {
        if (!result.savedProfiles) return;

        // since you're using an object, you can store it directly in chrome storage without any transformation required
        setProfiles(result.savedProfiles);
    });
}, []);
```